walk-in-interview/src/main/java/com/google/job/servlets/JobsListingsServlet.java

```diff
@@ -0,0 +1,228 @@
+
+import java.sql.Time;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.*;
+
+/**
+ * Servlet that handles getting the job listings given the filters.
+ */
+@WebServlet("/jobs/listings")
+public final class JobsListingsServlet extends HttpServlet {
+    private static final long TIMEOUT_SECONDS = 5;
+
+    private static final String MIN_LIMIT_PARAM = "minLimit";
+    private static final String MAX_LIMIT_PARAM = "maxLimit";
+    private static final String REGION_PARAM = "region";
+    private static final String SORT_BY_PARAM = "sortBy";
+    private static final String ORDER_PARAM = "order";
+    private static final String PAGE_SIZE_PARAM = "pageSize";
+    private static final String PAGE_INDEX_PARAM = "pageIndex";
+
+    private JobsDatabase jobsDatabase;
+
+    @Override
+    public void init() {
+        this.jobsDatabase = new JobsDatabase();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        try {
+            
+            JobQuery jobQuery = parseJobQuery(request);
+
+            JobPage jobPage = fetchJobPageDetails(jobQuery);
+
+            String json = ServletUtils.convertToJsonUsingGson(jobPage);
+            response.setContentType("application/json;");
+            response.getWriter().println(json);
+        } catch(IllegalArgumentException | ServletException | ExecutionException | TimeoutException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    /**
+     * Returns the JobPage object.
+     *
+     * @param jobQuery The job query object with all the filtering/sorting params.
+     * @return JobPage object with all the details for the GET response.
+     */
+    private JobPage fetchJobPageDetails(JobQuery jobQuery) throws ServletException, ExecutionException, TimeoutException {
+        try {
+            return this.jobsDatabase.fetchJobPage(jobQuery)
+                    .get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (InterruptedException | IOException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    /**
+     * Returns the job query object.
+     *
+     * @param request From the GET request.
+     * @return the job query object.
+     * @throws IllegalArgumentException if one of the params is invalid.
+     */
+    public static JobQuery parseJobQuery(HttpServletRequest request) throws IllegalArgumentException {
+        int minLimit = parseMinLimit(request);
+        int maxLimit = parseMaxLimit(request);
+        SingaporeRegion region = parseRegion(request);
+        Filter sortBy = parseSortBy(request);
+        Order order = parseOrder(request);
+        // TODO(issue/34): parse page size and index once pagination is implemented
+
+        JobQuery jobQuery = new JobQuery().setMinLimit(minLimit).setMaxLimit(maxLimit).setRegion(region)
```

Comment:
nit: i feel this should have been formatted. Typically it would look like this (when formatted automatically):
```
JobQuery jobQuery = new JobQuery()
                                           .setMinLimit(minLimit)
                                           .setMaxLimit(minLimit)
[...]
```

also you can use return directly:
```
return new JobQuery()
                   .setMinLimit(minLimit)
                    .setMaxLimit(minLimit)
[...]
```